pennylane/templates/state_preparations/state_prep_mps.py

```diff
@@ -15,24 +15,124 @@
 Contains the MPSPrep template.
 """
 
+import numpy as np
+
 import pennylane as qml
 from pennylane.operation import Operation
 from pennylane.wires import Wires
 
 
+def right_canonicalize_mps(mps):
+    r"""Transform a matrix product state into its right-canonical form.
+
+    Args:
+        mps (list[TensorLike]): List of tensors representing the MPS.
+
+    Returns:
+        A list of tensors representing the MPS in right-canonical form.
+
+    .. seealso:: :class:`~.MPSPrep`.
+
+    A right-canonicalized MPS is a matrix product state where each tensor :math:`A^{(j)}` satisfies
+    the following orthonormality condition:
+
+    .. math::
+
+        \sum_{d_{j,1}, d_{j,2}} A^{(j)}_{d_{j, 0}, d_{j, 1}, d_{j, 2}} \left( A^{(j)}_{d'_{j, 0}, d_{j, 1}, d_{j, 2}} \right)^* = \delta_{d_{j, 0}, d'_{j, 0}}
+
+    **Example**
+
+    .. code-block::
+
+        n_sites = 4
+
+        mps = (
+            [np.ones((1, 2, 4))]
+            + [np.ones((4, 2, 4)) for _ in range(1, n_sites - 1)]
+            + [np.ones((4, 2, 1))]
+        )
+        mps_rc = qml.right_canonicalize_mps(mps)
+
+        for i in range(1, n_sites - 1):
+            tensor = mps_rc[i]
+
+            # Right-canonical definition
+            contraction_matrix = np.tensordot(tensor, tensor.conj(), axes=([1, 2], [1, 2]))
+            print(np.allclose(contraction_matrix, np.eye(tensor.shape[0])))
+
+    .. code-block:: pycon
+
+        True
+        True
+    """
+
+    is_right_canonical = True
+    d_shapes = []
+    for tensor in mps[1:-1]:
+        # Right-canonical definition
+        d_shapes += tensor.shape
+        input_matrix = qml.math.tensordot(tensor, tensor.conj(), axes=([1, 2], [1, 2]))
+        if not qml.math.allclose(input_matrix, qml.math.eye(tensor.shape[0])):
+            is_right_canonical = False
+
+    if is_right_canonical:
+        return mps
```

Comment:
This logic looks a bit strange to me. This is what I think:
1. You are checking if input mps is in right_canonical.
2. If yes, return mps.
3. If not, perform right canonicalization.

If this is correct, then you can just use 

```
if qml.math.allclose(...):
    return mps
```

Reply:
I can't do what you say because I have to check that indeed all the tensors of the mps satisfy the definition

I have to check `qml.math.allclose(...)` for each tensor